### PR TITLE
#162414160 Send verification link via email upon successful signup

### PR DESCRIPTION
**Pull request by iMichaelOwolabi**
#### What does this PR do?
Send a verification link to user via email after a successful registration using Sendgrid mail service

#### Description of Task to be completed?
After a successful user registration, a verification livk is sent to users email to verify their account

#### How should this be manually tested?
* Pull the current repository
* Run npm install to install all dependencies.
* run npm test
* An email will be sent to the users email

#### Screenshots (if appropriate)
screenshot 2019-01-18 at 10 49 14 am
screenshot 2019-01-18 at 10 42 17 am

#### Questions:
What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162414160

### CODE REVIEW

File: controllers/verificationEmailController.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const { verifyToken } = require('../lib/utils/jwtUtils');
+const User = require('../lib/modelManagers/usermodel');
+/**
+ *
+ *
+ * @param {*} req
+ * @param {*} res
+ * @returns {object} response object based on the user account verification operation.
+ */
+async function verifyNewUser(req, res) {
```
Comment (codrex) [inline controllers/verificationEmailController.js]: Please refactor this function.

File: lib/utils/emailService/emailVerification.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    subject: SUBJECT,
+    html: emailTemplate(hostUrl)
+  };
+
+  try {
+    const sendMail = await sgMail.send(msg);
+    if (sendMail) {
+      return res.status(201).json({
+        success: true,
+        message: 'Account successfully created, Kindly check your email to activate your account',
```
Comment (codrex) [inline lib/utils/emailService/emailVerification.js]: move to constants file

File: lib/utils/jwtUtils.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const jwt = require('jsonwebtoken');
```
Comment (codrex) [inline lib/utils/jwtUtils.js]: Why did you create this file? You have exceeded the scope of your work

File: test/emailVerificationTest.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const { expect } = require('chai');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+const sendVerificationMail = require('../lib/utils/emailService/emailVerification');
+const verifyNewUser = require('../controllers/verificationEmailController');
+const User = require('../lib/modelManagers/usermodel');
+
+const validToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1dWlkIjoiNDE1ZTJlZmEtMTllNy0xMWU5LWFiMTQtZDY2M2JkODczZDkzIiwiaWF0IjoxNTQ3ODIxMjk2LCJleHAiOjE1NDc5MDc2OTZ9.DVwT7E9aU92ByHEw7GNn8URI-iZAR2VCLIY47dilOec';
```
Comment (codrex) [inline test/emailVerificationTest.js]: move to constants file

File: test/emailVerificationTest.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      message: 'Account successfully created, Kindly check your email to activate your account'
+    },
+  };
+  it('Should send verification mail to new user', async () => {
+    await sendVerificationMail(req, res);
+    expect(res.json.success).to.equal(true);
+  });
+});
+
+describe('GET /api/v1/auth/verifyemail/:token', () => {
```
Comment (codrex) [inline test/emailVerificationTest.js]: `'GET /api/v1/auth/verifyemail/:token'` are you testing an endpoint?

File: test/emailVerificationTest.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const { expect } = require('chai');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+const sendVerificationMail = require('../lib/utils/emailService/emailVerification');
```
Comment (codrex) [inline test/emailVerificationTest.js]: `sendVerificationMail` and `verifyNewUser ` should be tested in separate files

Comment (iMichaelOwolabi) [reply test/emailVerificationTest.js]: > `sendVerificationMail` and `verifyNewUser ` should be tested in separate files

I will separate them immediately

File: lib/utils/jwtUtils.js (hunk 3 above)
Comment (iMichaelOwolabi) [reply lib/utils/jwtUtils.js]: fixed

File: test/emailVerificationTest.js (hunk 5 above)
Comment (iMichaelOwolabi) [reply test/emailVerificationTest.js]: fixed

File: controllers/verificationEmailController.js (hunk 1 above)
Comment (iMichaelOwolabi) [reply controllers/verificationEmailController.js]: done

File: lib/utils/emailService/emailVerification.js (hunk 2 above)
Comment (iMichaelOwolabi) [reply lib/utils/emailService/emailVerification.js]: done

File: test/emailVerificationTest.js (hunk 4 above)
Comment (iMichaelOwolabi) [reply test/emailVerificationTest.js]: done

**Review by codrex — APPROVED** [review]

**Pull request merged**